runtime-modules/project-token/src/lib.rs

```diff
@@ -800,6 +801,239 @@ decl_module! {
+            }
+
+            Self::deposit_event(RawEvent::BondingCurveActivated(token_id, member_id, curve));
+
+            Ok(())
+        }
+
+        /// Activate Amm functionality for the token
+        /// Preconditions
+        /// - origin, member_id pair must be a valid authentication pair
+        /// - token_id must exist
+        /// - user usable JOY balance must be enough for bonding (+ existential deposit eventually)
+        /// - deadline constraint respected if provided
+        /// - slippage tolerance constraints respected if provided
+        /// - token total supply and amount value must be s.t. `eval` function doesn't overflow
+        ///
+        /// Postconditions
+        /// - token `bonding_curve` activated with specified parameters
+        /// - `amount` CRT minted into account (which is created if necessary and existential deposit added to it)
+        /// - respective JOY amount transferred from user balance to amm treasury account
+        /// - percentage of the minted CRT transferred to the token content creator (creator reward)
+        /// - event deposited
+        #[weight = 100_000_000] // TODO: adjust weight
+        fn bond(origin, token_id: T::TokenId, member_id: T::MemberId, amount: <T as Config>::Balance, deadline: Option<<T as timestamp::Config>::Moment>, slippage_tolerance: Option<(Permill, JoyBalanceOf<T>)>) -> DispatchResult {
+            if amount.is_zero() {
+                return Ok(()); // noop
+            }
+
+            let sender = ensure_signed(origin.clone())?;
+
+            T::MemberOriginValidator::ensure_member_controller_account_origin(
+                origin,
+                member_id
+            )?;
+
+            let token_data = Self::ensure_token_exists(token_id)?;
+            let curve = token_data.bonding_curve.ok_or(Error::<T>::NotInAmmState)?;
+
+            let user_account_data_exists = AccountInfoByTokenAndMember::<T>::contains_key(token_id, &member_id);
+            let amm_reserve_account = Self::module_bonding_curve_reserve_account(token_id);
+            let price = curve.eval::<T>(amount, token_data.total_supply, BondOperation::Bond)?;
+            let bloat_bond = Self::bloat_bond();
+
+            let joys_required = if !user_account_data_exists {
+                price.saturating_add(bloat_bond)
+            } else {
+                price
+            };
+
+            Self::ensure_can_transfer_joy(&sender, joys_required)?;
+
+            // slippage tolerance check
+            if let Some((slippage_tolerance, desired_price)) = slippage_tolerance {
+                ensure!(price.saturating_sub(desired_price) < slippage_tolerance.mul_floor(desired_price), Error::<T>::SlippageToleranceExceeded);
+            }
+
+            // timestamp deadline check
+            if let Some(deadline) = deadline {
+                ensure!(<timestamp::Pallet<T>>::get() < deadline, Error::<T>::DeadlineExpired);
```

Comment:
```suggestion
                ensure!(<timestamp::Pallet<T>>::get() <= deadline, Error::<T>::DeadlineExpired);
```
(same as with `slippage_tolerance`, for consistency)